File: treinos/treino026/app/src/main/java/aluno/gabriel/treino026/view/SplashActivity.java
package aluno.gabriel.treino026.view;

import androidx.appcompat.app.AppCompatActivity;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.widget.TextView;

import aluno.gabriel.treino026.BuildConfig;
import aluno.gabriel.treino026.R;

public class SplashActivity extends AppCompatActivity {

    TextView txtNomeApp;
    TextView txtVersaoApp;
    String versionApp;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);

        includeElementsById();
        replaceElements();
        transferenceScreen();
    }

    protected void transferenceScreen(){
        Boolean handler = new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(SplashActivity.this, MainActivity.class);
                startActivity(intent);
                finish();
            }
        },1000);
    }

    protected void includeElementsById(){
        txtNomeApp = findViewById(R.id.txtNomeApp);
        txtVersaoApp = findViewById(R.id.txtVersaoApp);
    }

    @SuppressLint("SetTextI18n")
    protected void replaceElements(){
        versionApp = BuildConfig.VERSION_NAME;
        txtVersaoApp.setText("Versão: "+versionApp);

    }
}

File: treinos/treino027/app/src/main/java/aluno/gabriel/treino027/view/SplashActivity.kt
package aluno.gabriel.treino027.view

import aluno.gabriel.treino027.R
import aluno.gabriel.treino027.core.AppUtil
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.widget.TextView

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        trocarTela()
        getApplicationInformations()
    }

    fun trocarTela(){
        Handler(Looper.getMainLooper()).postDelayed({
            startActivity(Intent(this@SplashActivity, LoginActivity::class.java))
            finish()
        },1000)
    }

    fun getApplicationInformations(){
        val txtVersaoApp: TextView = findViewById(R.id.txtVersaoApp)
        val txtNomeApp: TextView = findViewById(R.id.txtNomeApp)

        txtNomeApp.text = ""+AppUtil.Companion.APPLICATION_NAME
        txtVersaoApp.text = "Versão: "+ AppUtil.Companion.APPLICATION_VERSION

    }
}